docs/reference/strava-model.rst

```diff
@@ -0,0 +1,70 @@
+============================
+Strava_model Module
+============================
+.. currentmodule:: stravalib.strava_model
+
+The `strava-model` module provides... stuff and lots of it
```

Comment:
Wanted to chat a bit about organizing the API section. sometimes we have a huge list of classes. We could organize them by themes such as activity, map / spatial, streaming etc... but we'd want to agree on an organization that will make sense to users. otherwise for now we can include as a big list and think about better organization for the future? open to either.

Reply:
I'm not sure it makes sense to include `strava_model` here since it will not be directly used or edited by contributors. It could be that I added it in #324, but I start to doubt its usefulness. Because `strava_model` is generated, we'll need to keep track of its changes in this file too.

Not sure if it's possible, but it would be nice if the classes in `strava_model` can be seen in the docs as just the superclasses of our entities in `model`.

Reply:
> I'm not sure it makes sense to include strava_model here since it will not be directly used or edited by contributors. It could be that I added it in https://github.com/stravalib/stravalib/pull/324, but I start to doubt its usefulness. Because strava_model is generated, we'll need to keep track of its changes in this file too.

One option here to avoid keeping track of it is to just use  the `automodule`. if we do that and i don't manually add all of the functions - the page will just automagically build and rebuild as the .py file is updated. maybe we can just do it that way? no maintenance but the module still has a nice docs page? 

it would look like this then:

```
=============
Strava model
=============

.. automodule:: stravalib.strava_model
   :members:
   :undoc-members:
```
One potentially glitch might be the menu might get a bit long but i can play with that. 

> Not sure if it's possible, but it would be nice if the classes in strava_model can be seen in the docs as just the superclasses of our entities in model.

How would you want to represent strava_model as a superclass in the docs? what would the desired output look like? I understand conceptually (i think) what you are saying - but am not sure what it would look like in the docs.

Reply:
> How would you want to represent strava_model as a superclass in the docs? what would the desired output look like?

The `model.Club` API doc now looks like this:

<img width="792" alt="stravalib_model_Club_-_Stravalib_Docs" src="https://user-images.githubusercontent.com/1788027/216608304-b5e882e5-61be-4b8b-9346-d5fcf92b9882.png">

You can see that only the undocumented fields that we explicitly added to this class are shown. However, `model.Club` inherits from `strava_model.DetailedClub`, and as such has a lot more fields:

```python
class DetailedClub(SummaryClub):
    admin: Optional[bool] = None
    """
    Whether the currently logged-in athlete is an administrator of this club.
    """
    following_count: Optional[int] = None
    """
    The number of athletes in the club that the logged-in athlete follows.
    """
    membership: Optional[Literal["member", "pending"]] = None
    """
    The membership status of the logged-in athlete.
    """
    owner: Optional[bool] = None
    """
    Whether the currently logged-in athlete is the owner of this club.
    """
```

and, of course, the fields that `strava_model.DetailedClub` inherits from `strava_model.SummaryClub`, etc.

I just added a commit that should resolve this.

Reply:
BTW, using `automodule` for `strava_model` is fine.